contract_manager/src/contracts/sui.ts

```diff
@@ -185,18 +192,114 @@ export class SuiContract extends Contract {
   async executeMigrateInstruction(vaa: Buffer, keypair: Ed25519Keypair) {
     const tx = new TransactionBlock();
     const packageId = await this.getPythPackageId();
-    let decreeReceipt = await this.getVaaDecreeReceipt(tx, packageId, vaa);
+    let verificationReceipt = await this.getVaaVerificationReceipt(
+      tx,
+      packageId,
+      vaa
+    );
 
     tx.moveCall({
       target: `${packageId}::migrate::migrate`,
-      arguments: [tx.object(this.stateId), decreeReceipt],
+      arguments: [tx.object(this.stateId), verificationReceipt],
     });
 
     return this.executeTransaction(tx, keypair);
   }
 
   async executeUpdatePriceFeed(senderPrivateKey: string, vaas: Buffer[]) {
-    throw new Error("Not implemented");
+    throw new Error("Use executeUpdatePriceFeedWithFeeds instead");
+  }
+
+  async verifyVaas(vaas: Buffer[], tx: TransactionBlock) {
+    const wormholePackageId = await this.getWormholePackageId();
+    const verifiedVaas = [];
+    for (const vaa of vaas) {
+      const [verifiedVaa] = tx.moveCall({
+        target: `${wormholePackageId}::vaa::parse_and_verify`,
+        arguments: [
+          tx.object(this.wormholeStateId),
+          tx.pure([...vaa]),
+          tx.object(SUI_CLOCK_OBJECT_ID),
+        ],
+      });
+      verifiedVaas.push(verifiedVaa);
+    }
+    return verifiedVaas;
+  }
```

Comment:
Can we reuse the login on the sui sdk?

Reply:
Yeah, that was the plan after merging the other PR